bowling/src/input.cpp

```
@@ -0,0 +1,65 @@
+#include <algorithm>
+#include <fstream>
+
+#include "input.hpp"
+
+namespace fs = std::experimental::filesystem;
+
+Input::Input(const std::string& directory) : directoryPath_(fs::path(directory)), isValid_(false) {
+    if (!fs::exists(directoryPath_)) {
+        return;
+    }
+    if (!fs::is_directory(directoryPath_)) {
+        return;
+    }
```

consider if better to throw an exception.

---

and move it into private method `checkDirectory()`

---

I prefer not to throw exception. moved to function.

---

I think that exception is made for place like this, and we should use it as well

---

We can leave without exception here